core/src/main/clojure/xtdb/core/datalog.clj

```diff
@@ -19,7 +19,7 @@
 (s/def ::eid ::lp/value)
 (s/def ::attr keyword?)
 (s/def ::value ::lp/value)
-(s/def ::table simple-symbol?)
+(s/def ::table (s/and simple-keyword? (s/conformer #(symbol (name %)) #(keyword (name %)))))
```

Comment:
might just need `(s/conformer symbol keyword)` here - IIRC `(symbol :foo)` and `(keyword 'bar)` both work?